Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -672,12 +672,21 @@ PhysicalParticleContainer::AddPlasma (int lev, RealBox part_realbox)
                 // If the particle is not within the species's
                 // xmin, xmax, ymin, ymax, zmin, zmax, go to
                 // the next generated particle.
-                if (!inj_pos->insideBounds(xb, yb, z)) {
+
+                // include ballistic correction for plasma species with bulk motion
+                // note that this assumes that momentum profile is independent of z
+                u = inj_mom->getBulkMomentum(x, y, 0.);
+                Real gamma = std::sqrt( 1.+(u.x*u.x+u.y*u.y+u.z*u.z) );
+                Real betaz = u.z/gamma;
+                Real z0 = z - PhysConst::c*t*betaz;
```

Comment:
Could we append these variables with `_bulk`? Or would that be too many code changes?